### ISSUE TITLE
Look into failing test cases v0.1.4

### ISSUE THREAD

**Issue by shubhra-agrawal** [ElucidataInc/ElMaven#314]
Several test cases are failing in v0.1.4 which need to be fixed.

**Comment by shubhra-agrawal** [comment]
@chubukov There were 8 test cases that were failing and this is how we have fixed them:

**eicMerge**- There was a bug in makeEICSlices while populating the vectors that led to the vectors being double their intended size. Reverting an old commit (#3bd3fb) fixed this.
**getEIC**- Passed by fixing eicMerge
**addGroup**- This function adds group information to CSV report. It was failing because of recent addition of formula column. The test case has been changed to reduce dependence on grouping algorithm therefore making it a better unit test.
**loadCompoundCSVFileWithRep**- Corner case test. If reference file has repetition, only 1 of the compound info is loaded. Fixing this required code change as the load count was getting incremented after every read instead of waiting for compound to be loaded.

All other tests required some values to be updated due to recent code changes. For example, addition of formula column in csv export.

Due to the changes in merged EIC and EIC slices, less number of groups are being reported compared to the current develop. We will merge it once we have tested it out properly.

**Comment by chubukov** [comment]
@shubhra-agrawal great. do you think it's likely that the first bug you mention was responsible for things like #333 and #334 ? Of course we have to test it out.

What was the original issue that was supposed to be addressed by 3bd3fb?

**Comment by shubhra-agrawal** [comment]
@chubukov #3bd3fb was done to improve memory utilization by changing from reserve() to resize() for memory allocation in vectors. Some other code changes that were required to successfully shift to resize() were not made and it ended up using even more memory than before. Basically, any functionality change wasn't intended. I've reverted that commit as mentioned and it has been merged to develop now. 

I have not been able to reproduce #333 and #334 yet but this change seems to have fixed #287 so it did affect grouping; probably because min RT was set to 0 for every EIC.
